src/axolotl/telemetry/manager.py

```diff
@@ -0,0 +1,252 @@
+        ) and do_not_track.lower() not in ("1", "true")
+
+        # Show warning (and sleep on all ranks) unless explicitly enabled
+        if enabled and not explicit_enabled:
+            if is_main_process():
+                LOG.warning(ENABLED_WARNING)
+            time.sleep(ENABLED_WARNING_SLEEP_SECONDS)
+
+        # Only rank 0 will send telemetry
+        if not is_main_process():
+            return False, False
+
+        return enabled, explicit_enabled
+
+    def _load_whitelist(self) -> dict:
+        """Load HuggingFace Hub organization whitelist"""
+        with open(WHITELIST_PATH, encoding="utf-8") as f:
+            return yaml.safe_load(f)
+
+    def _is_whitelisted(self, base_model: str) -> bool:
+        """Check if model org is in whitelist"""
+        if not base_model:
+            return False
+
+        base_model = base_model.lower()
+        return any(
+            org.lower() in base_model for org in self.whitelist.get("organizations", [])
+        )
+
+    def _init_posthog(self):
+        """Initialize PostHog client"""
+        posthog.host = POSTHOG_HOST
+        posthog.project_api_key = POSTHOG_WRITE_KEY
+
+    def _redact_paths(self, properties: dict[str, Any]) -> dict[str, Any]:
+        """
+        Redact properties to remove any paths, so as to avoid inadvertently collecting
+        private or personally identifiable information (PII).
+
+        Args:
+            properties: Dictionary of properties to redact.
+
+        Returns:
+            Properties dictionary with paths redacted.
+        """
+        if not properties:
+            return {}
+
+        # NOTE: Keep this up to date with any config schema changes
+        path_indicators = {"path", "dir"}
+
+        def redact_value(value: Any, key: str = "") -> Any:
+            """Recursively sanitize values, redacting those with path-like keys"""
+            # Special case: base_model should be redacted if org is not whitelisted
+            if key == "base_model":
+                org = value.split("/")[0]
+                if org not in self.whitelist["organizations"]:
+                    return "[REDACTED]"
+
+            if isinstance(value, str):
+                # If the key suggests this is a path, redact it
+                if any(indicator in key.lower() for indicator in path_indicators):
+                    return "[REDACTED]"
+
+            # Handle nested structures
+            if isinstance(value, dict):
+                return {k: redact_value(v, k) for k, v in value.items()}
+            if isinstance(value, list):
+                return [redact_value(item) for item in value]
+
+            return value
+
+        # Create new dict with redacted values
+        redacted = {k: redact_value(v, k) for k, v in properties.items()}
+
+        return redacted
+
+    def _get_system_info(self) -> dict[str, Any]:
+        """Collect system information"""
+        gpu_info = []
+        if torch.cuda.is_available():
```

Comment:
Do we need to consider for NPU/MPS?

Reply:
Good point!

I wonder if we should setup testing for different accelerators in our CI/CD. Otherwise, I will have to test all the possible hardware manually, which I'm not looking forward to 😅 @winglian any thoughts here?